cmake/CMakeLists.txt

```diff
@@ -30,6 +30,8 @@ set(CMAKE_CXX_STANDARD 14)
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 # NOTE: POSITION INDEPENDENT CODE hurts performance, and it only make sense on POSIX systems
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+# Generate build logs listing all flags used
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```

Comment:
This  will affect all the builds, no matter it is windows, linux or mac.

Reply:
Yes it is, but all it does is to generate a json file of build commands ever issued, no impact on output binaries or build time.

Reply:
FYI, I set this from build.sh:
`build.sh ... --cmake_extra_defines CMAKE_EXPORT_COMPILE_COMMANDS=1`

That's another way to do it if we don't want it on by default.

Reply:
Can we do this for Mac only since the PIE requirement was for Mac only?

Reply:
Done - moved to "if Darwin...".